src/test/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchServiceTest.kt

```diff
@@ -0,0 +1,179 @@
+package uk.gov.justice.hmpps.offendersearch.services
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.nhaarman.mockito_kotlin.any
+import com.nhaarman.mockito_kotlin.check
+import com.nhaarman.mockito_kotlin.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.elasticsearch.action.search.SearchRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.action.search.SearchResponseSections
+import org.elasticsearch.client.RestHighLevelClient
+import org.elasticsearch.common.bytes.BytesArray
+import org.elasticsearch.index.query.BoolQueryBuilder
+import org.elasticsearch.index.query.MatchQueryBuilder
+import org.elasticsearch.search.SearchHit
+import org.elasticsearch.search.SearchHits
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.mockito.Mock
+import org.mockito.Mockito.verify
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import uk.gov.justice.hmpps.offendersearch.dto.MatchRequest
+import uk.gov.justice.hmpps.offendersearch.dto.OffenderDetail
+import java.time.LocalDate
+
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.MOCK)
```

Comment:
I have had to make this a spring boot test which is OTT, when all I need is an objectMapper that works with a Kotlin data class. I tried everything, but would always fail to marshal data into a data class (no constructor error) unless  i used the one from Spring. Any advice would be appreciated.

Reply:
did you try adding https://github.com/FasterXML/jackson-module-kotlin?  Normally if you don't have it there will be a warning on startup

Reply:
Thanks, I'll have a go at that